### PR TITLE
Use os matrix and simplify tests

### PR DESCRIPTION
**Pull request by stan-sz**
* Add Linux builds to PR gate
* simplify test to use data-driven attributes
* copy the props and targets for each test case instead of generating them

### CODE REVIEW

File: test/E2ETests.cs [hunk 1]
```diff
@@ -7,6 +7,8 @@
 namespace ReferenceTrimmer.Tests;
 
 [TestClass]
+[DeploymentItem("TestData/Directory.Build.props")]
```
Comment (dfederm) [inline test/E2ETests.cs:10]: Isn't deployment item legacy / deprecated? You should instead just have them as content and read the files as if they were relative to the cwd

Comment (stan-sz) [reply test/E2ETests.cs:10]: There's no official info about it being legacy, and as you looks at the test code, it's just easier to read. Let's poke around for other ways to make it declarative.

Comment (dfederm) [reply test/E2ETests.cs:10]: How about a data driven test where the data source is a property which provides the test name and the expectations? I forgot the name offhand but "data driven testing" in the docs is what to look for. "DataTestMethod" I think?

Comment (dfederm) [reply test/E2ETests.cs:10]: Or just leave it as-is. I'm not convinced the existing pattern is even bad in any way.

**Pull request closed without merging**